cmd/livepeer_cli/wizard_ticketbroker.go

```diff
@@ -0,0 +1,217 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"net/http"
+	"net/url"
+
+	"github.com/golang/glog"
+	"github.com/livepeer/go-livepeer/eth"
+)
+
+func (w *wizard) deposit() {
+	sender, err := w.senderInfo()
+	if err != nil {
+		glog.Errorf("Error getting sender info: %v", err)
+		return
+	}
+
+	params, err := w.ticketBrokerParams()
+	if err != nil {
+		glog.Errorf("Error getting TicketBroker params: %v", err)
+		return
+	}
+
+	reqPenaltyEscrowAmount := big.NewInt(0)
+	if params.MinPenaltyEscrow.Cmp(sender.PenaltyEscrow) > 0 {
+		reqPenaltyEscrowAmount = new(big.Int).Sub(params.MinPenaltyEscrow, sender.PenaltyEscrow)
+	}
+
+	if reqPenaltyEscrowAmount.Cmp(big.NewInt(0)) > 0 {
+		w.fundAndApproveSigners(sender.Deposit, sender.PenaltyEscrow, reqPenaltyEscrowAmount)
+		return
+	}
+
+	w.fundDeposit(sender.Deposit)
+
+	return
+}
+
+func (w *wizard) fundAndApproveSigners(currDeposit *big.Int, currPenaltyEscrow *big.Int, reqPenaltyEscrowAmount *big.Int) {
+	fmt.Printf("Current Deposit: %v\n", currDeposit)
+	fmt.Printf("Current Penalty Escrow: %v\n", currPenaltyEscrow)
+
+	fmt.Printf("Enter deposit amount in ETH (%v will be used cover the minimum penalty escrow) - ", eth.FormatUnits(reqPenaltyEscrowAmount, "ETH"))
+
+	amount := w.readFloatAndValidate(func(in float64) (float64, error) {
+		amountWei := eth.ToBaseUnit(big.NewFloat(in))
+
+		if amountWei.Cmp(reqPenaltyEscrowAmount) < 0 {
+			return 0, fmt.Errorf("must deposit at least %v to cover the minimum penalty escrow", eth.FormatUnits(reqPenaltyEscrowAmount, "ETH"))
+		}
+
+		return in, nil
+	})
+
+	form := url.Values{
+		"amount": {eth.ToBaseUnit(big.NewFloat(amount)).String()},
+	}
+	fmt.Println(httpPostWithParams(fmt.Sprintf("http://%v:%v/fundAndApproveSigners", w.host, w.httpPort), form))
+}
+
+func (w *wizard) fundDeposit(currDeposit *big.Int) {
+	fmt.Printf("Current Deposit: %v\n", currDeposit)
+
+	fmt.Printf("Enter deposit amount in ETH - ")
+
+	amount := w.readFloat()
+
+	form := url.Values{
+		"amount": {eth.ToBaseUnit(big.NewFloat(amount)).String()},
+	}
+	fmt.Println(httpPostWithParams(fmt.Sprintf("http://%v:%v/fundDeposit", w.host, w.httpPort), form))
+}
+
+func (w *wizard) unlock() {
+	sender, err := w.senderInfo()
+	if err != nil {
+		glog.Errorf("Error getting sender info: %v", err)
+		return
+	}
+
+	if sender.WithdrawBlock.Cmp(big.NewInt(0)) > 0 {
+		fmt.Println("Already in the unlock period")
+		return
+	}
+
+	fmt.Printf("Current Deposit: %v\n", eth.FormatUnits(sender.Deposit, "ETH"))
+	fmt.Printf("Current Penalty Escrow: %v\n", eth.FormatUnits(sender.PenaltyEscrow, "ETH"))
+
+	fmt.Printf("Would you like initiate the unlock period? (y/n) - ")
+
+	input := w.readStringAndValidate(func(in string) (string, error) {
+		if in != "y" && in != "n" {
+			return "", errors.New("Enter (y)es or (n)o")
```

Comment:
I feel like this is slightly confusing.. perhaps we can change the instructions or the values we accept to have a more straightforward relationship between instruction and parsing?

for example we could ask the user only for `y / n` in the print out OR we could accept "yes" / "no" strings.

wdyt?

Reply:
another note - I think it's a worthwhile refactor to just extract the multiple copies of the y/n validation implementation to a single func and just provide a pointer to that func here, to promote reuse.

wdyt?

Reply:
> for example we could ask the user only for y / n in the print out OR we could accept "yes" / "no" strings.

For the error message, I was trying to be super obvious by indicating that y = yes and n = no. We could also just say `Enter y or n`

Reply:
Updated to use the [readStringYesOrNo()](https://github.com/livepeer/go-livepeer/pull/658/files#diff-5d52c4c551172dd0fedcbb693fdda67eR64) helper